### ISSUE TITLE
GQL form submission: I'm able to submit a disabled entry

### ISSUE THREAD

**Issue by tremby** [verbb/formie#874]
**Description**

Even with a somewhat restrictive GQL schema (allowing submission and reading of forms, but no modification or deletion) I'm able to submit entries marked as "disabled".

**Steps to reproduce**
1. Submit a form via GQL. Set `enabled: false` as part of the mutation.
2. See that no result comes back. This looks like it was denied, though there was no error message saying so.
3. Check submissions, see that a new entry, in disabled state, was added.

Users should not be able to set their entry to disabled without some kind of special permission, if at all.

### Form settings
Please pick from the options below.

- Multi-page form: n/a
- Submission Method: GraphQL
- Client-side Validation: n/a
- Custom Form Templates: n/a

**Additional info**
- Plugin version: 1482b90ba
- Craft version: 4.0.0.1
- Multi-site: Yes

**Comment by engram-design** [comment]
Can you let me know your schema permissions?

- Edit submissions for form “{form}”
    - Create submissions for form “{form}”
    - Modify submissions for form “{form}”
    - Delete submissions for form “{form}”

Just trying to compare the same behaviour with entries and other Craft elements. I assume you have Edit/Create enabled?

**Comment by tremby** [comment]
- View all forms on
- Edit all forms on (just to reveal the next checkboxes)
  - Create all submissions on
  - Modify all submissions off
  - Delete all submissions off
- Everything else related to formie off

**Comment by engram-design** [comment]
So to me, I feel like this is expected behaviour. The permissions allow you to create a submission - any submission, whether it be disabled, enabled, or of a different status. This mirrors what other elements like Entries do with the same permissions.

```
mutation saveEntry($title: String) {
  save_gql_gql_Entry(title: $title, enabled: false) {
    title
  }
}
```

Here, I can create an entry in Craft, even though it's disabled. We inherit this behaviour from how element's are dealt with (which we can change of course). The reason I mention it, is if we want this changed, I'd recommend we pitch it to Craft so other elements can benefit.

Totally understand the implications on setting the enabled/status on elements via GraphQL, and it opens up parties to doing whatever they like. But there's also legitimate reasons to create disabled submissions, or submissions of a non-default status.

As such, I'd say the solution for this is maybe to think about ways to tighten the schema permissions about _what_ new element can be created. I'm not quite sure how to handle that in a nice way, but I think that might be nice for all elements.

Maybe some additional checkboxes for:
- Edit submissions for form “{form}”
    - Create submissions for form “{form}”
        - Create disabled submissions
        - Create non-default status submissions
    - Modify submissions for form “{form}”

Or something similar?

**Comment by tremby** [comment]
That sounds reasonable.

Honestly I'm still new to Craft and I don't know what kind of other entries people make. I suppose I was under the impression that "create entry" would only be enabled for private schemas (like for server-side parts of applications) and would always be switched off for public schemas; you probably wouldn't mind all the knobs being available in this situation, but this would put form entries in a different category.

Can you give an example of a sort of entry you might want anyone in the public to have access to create? (Maybe my imagination is broken.)

**Comment by engram-design** [comment]
Yeah, Craft entries are different to Formie Submissions, as they can be used for lots of different cases. But the same thing applies to Categories, Users, Products, etc - all elements. 

Lots of different scenarios, but anything user-content created. Maybe a site runs a way to create campaigns or articles from the front-end. It'd be valid to allow users to create articles form the front-end, using the GQL API. Sure, so long as anyone with the tokenId can perform this action, I would lock-down the API a little more to ensure only requests from a specific origin could come through.

But getting back to the forms side of things, I'll consider those permissions
